src/main/java/com/pnu/dev/shpaltaif/repository/specification/PostSpecificationBuilder.java

```diff
@@ -12,43 +12,63 @@
 import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import static java.util.Objects.nonNull;
 
 @Component
 public class PostSpecificationBuilder {
 
     private final PublicAccountService publicAccountRepository;
 
-    private final CategoryService categoryRepository;
+    private final CategoryService categoryService;
 
     @Autowired
     public PostSpecificationBuilder(PublicAccountService publicAccountService, CategoryService categoryService) {
         this.publicAccountRepository = publicAccountService;
-        this.categoryRepository = categoryService;
+        this.categoryService = categoryService;
     }
 
     public Specification<Post> buildPostSpecification(User user, PostFiltersDto postFiltersDto) {
 
-        PostSpecification specification = new PostSpecification();
-
-        specification.add(new SearchCriteria("active", postFiltersDto.isActive(), SearchOperation.EQUAL));
+        List<SearchCriteria> searchCriteriaList = buildCommonSearchCriteriaList(postFiltersDto);
 
         if (user.getRole().equals(UserRole.ROLE_WRITER)) {
-            specification.add(new SearchCriteria("authorPublicAccount", user.getPublicAccount(), SearchOperation.EQUAL));
+            searchCriteriaList.add(new SearchCriteria("authorPublicAccount", user.getPublicAccount(), SearchOperation.EQUAL));
         } else if (nonNull(postFiltersDto.getAuthorPublicAccountId())) {
             PublicAccount publicAccount = publicAccountRepository.findById(postFiltersDto.getAuthorPublicAccountId());
-            specification.add(new SearchCriteria("authorPublicAccount", publicAccount, SearchOperation.EQUAL));
+            searchCriteriaList.add(new SearchCriteria("authorPublicAccount", publicAccount, SearchOperation.EQUAL));
         }
 
+        return new PostSpecification(searchCriteriaList);
+    }
+
+    public Specification<Post> buildPostSpecification(PostFiltersDto postFiltersDto) {
+
+        List<SearchCriteria> searchCriteriaList = buildCommonSearchCriteriaList(postFiltersDto);
+        return new PostSpecification(searchCriteriaList);
+    }
+
+    private List<SearchCriteria> buildCommonSearchCriteriaList(PostFiltersDto postFiltersDto) {
+
+        List<SearchCriteria> searchCriteriaList = new ArrayList<>();
+
+        searchCriteriaList.add(new SearchCriteria("active", postFiltersDto.isActive(), SearchOperation.EQUAL));
+
+
         if (nonNull(postFiltersDto.getTitle())) {
-            specification.add(new SearchCriteria("title", postFiltersDto.getTitle(), SearchOperation.MATCH));
+            searchCriteriaList.add(new SearchCriteria("title", postFiltersDto.getTitle(), SearchOperation.MATCH));
```

Comment:
For now we will perform search just by title, however can be changed in the future.